portality/view/openurl.py

```diff
@@ -0,0 +1,92 @@
+import re
+from flask import Blueprint, request, redirect, url_for, render_template, abort
+from portality.models import OpenURLRequest
+from urllib import unquote
+
+blueprint = Blueprint('openurl', __name__)
+
+@blueprint.route("/openurl", methods=["GET", "POST"])
+def openurl():
+
+    # Drop the first part of the url to get the raw query
+    url_query = request.url.split(request.base_url).pop()
+
+    # Validate the query syntax version and build an object representing it
+    parsed_object = parse_query(url_query)
+
+    # If it's not parsed to an OpenURLRequest, it's a redirect URL to try again.
+    if type(parsed_object) != OpenURLRequest:
+        return redirect(parsed_object, 301)
+
+    # Issue query and redirect to results page
+    results = parsed_object.query_es()
+    results_url = get_result_page(results)
+    if results_url:
+        return redirect(results_url)
+    else:
+        abort(400)
+
+def parse_query(url_query_string):
+    """
+    Create the model which holds the query
+    :param url_query_string: The query part of an incoming OpenURL request
+    :return: an object representing the query
+    """
+    # Check if this is new or old syntax, translate if necessary
+    match_1_0 = re.compile("url_ver=Z39.88-2004")
```

Comment:
Is that supposed to be a raw string match? If so I'd recommend just doing ```"url_ver=Z39.88-2004" in url_query_string``` on the line below rather than compile a RE. Performance is one reason, but more importantly things like the fact that the dot will match any character. So this should be ```"url_ver=Z39\.88-2004"``` as a RE.

Also if compiling RE-s, best to put them just under module imports up at the top. That way they'll be compiled on import and available for use throughout the lifetime of the web app thread.